### PR TITLE
Making snapshot name a scripted field

### PR DESCRIPTION
**Pull request by thalurur**
*Issue #, if available:*
https://discuss.opendistrocommunity.dev/t/universal-snapshot-policy-for-all-indexes/4815

*Description of changes:*
Making ISM policy snapshot name be deduced during runtime as a mustache script - this to expand support to use cases like making the snapshot name be the index its attached to

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt [hunk 1]
```diff
@@ -63,13 +68,12 @@ class AttemptSnapshotStep(
                 info = mutableInfo.toMap()
                 return this
             }
+            val snapshotNamePrefix = "-".plus(
```
Comment (dbbaughe) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt:71]: nit: Postfix

File: src/test/resources/mappings/cached-opendistro-ism-config.json [hunk 2]
```diff
@@ -194,7 +194,8 @@
                       "type": "keyword"
                     },
                     "snapshot": {
-                      "type": "keyword"
```
Comment (dbbaughe) [inline src/test/resources/mappings/cached-opendistro-ism-config.json:197]: Can't change the type of an existing mapping

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt (hunk 1 above)
Comment (thalurur) [reply src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/snapshot/AttemptSnapshotStep.kt:71]: right, will correct

**Comment by thalurur** [comment]
Closing in favor of - https://github.com/opensearch-project/index-management/pull/77

**Pull request closed without merging**